rts/Lua/LuaOpenGL.cpp

```diff
@@ -5661,5 +5666,69 @@ int LuaOpenGL::GetMapRendering(lua_State* L)
 	return 0;
 }
 
+
+/**
+ * @function gl.ObjectLabel labels an object for use with debugging tools
+ * @param objectTypeIdentifier GLenum Specifies the type of object being labeled.
+ * @param objectID GLuint Specifies the name or ID of the object to label.
+ * @param label string A null-terminated string containing the label to be assigned to the object.
+ * @treturn nil
+ */
+int LuaOpenGL::ObjectLabel(lua_State* L) {
+	GLenum identifier = (GLenum)luaL_checkinteger(L, 1);
+	if (identifier != GL_BUFFER && identifier != GL_SHADER && identifier != GL_PROGRAM &&
+		identifier != GL_VERTEX_ARRAY && identifier != GL_QUERY && identifier != GL_PROGRAM_PIPELINE &&
+		identifier != GL_TRANSFORM_FEEDBACK && identifier != GL_TEXTURE && identifier != GL_RENDERBUFFER &&
+		identifier != GL_FRAMEBUFFER) {
+		return 0;
+	}
+    GLuint objectID = (GLuint)luaL_checkinteger(L, 2);
+    const char* label = luaL_checkstring(L, 3);
+
+    // Ensure that identifier is a valid GLenum
+    // The length should be passed as -1 because the strings are null terminated
+    // Check that the GLuint name refers to a valid openGL object
+    // Check that the label is a valid string
+    glObjectLabel(identifier, objectID, -1, label);
+
+    return 0;
+}
+
+
+// https://registry.khronos.org/OpenGL-Refpages/gl4/html/glPushDebugGroup.xhtml
+/**
+ * @function gl.PushDebugGroup pushes a debug marker for nVidia nSight 2024.04, does not seem to work when FBO's are raw bound
+ * @param id GLuint A numeric identifier for the group.
+ * @param message string A human-readable string describing the debug group.
+ * @param source boolean true for GL_DEBUG_SOURCE_APPLICATION, false for GL_DEBUG_SOURCE_THIRD_PARTY. default false
+ * @treturn nil
+ */
+int LuaOpenGL::PushDebugGroup(lua_State* L) {
+	GLuint id = (GLuint)luaL_checkinteger(L, 1);
+	const char* message = luaL_checkstring(L, 2);
+	bool source = luaL_optboolean(L, 3, false);
+
+	GLint maxLength;
+	glGetIntegerv(GL_MAX_DEBUG_MESSAGE_LENGTH, &maxLength);
+
+	if (strlen(message) >= maxLength) {
+		luaL_error(L, "Message length exceeds GL_MAX_DEBUG_MESSAGE_LENGTH");
```

Comment:
Perhaps just trim the message to the appropriate length? Otherwise a portable way of preventing this error (`GL.MAX_DEBUG_MESSAGE_LENGTH` lua constant?) would be nice.